ts/src/renderer/phaser/classes/PhaserItem.ts

```diff
@@ -1,71 +1,35 @@
-class PhaserItem extends Phaser.GameObjects.Container {
-
-	sprite: Phaser.GameObjects.Sprite;
-
-	private playAnimationListener: EvtListener;
-	private hide: EvtListener;
-	private show: EvtListener;
-
-	constructor (scene: Phaser.Scene,
-				 private item: Item) {
-
-		super(scene);
-
-		const key = `item/${item._stats.itemTypeId}`;
-
-		const sprite = this.sprite = scene.add.sprite(0, 0, key);
-		this.add(sprite);
-
-		scene.add.existing(this);
-
-		this.hide = item.on('hide', () => {
-			this.sprite.setActive(false).setVisible(false);
-		});
-
-		this.show = item.on('show', () => {
-			this.sprite.setActive(true).setVisible(true);
-		});
-
-		this.playAnimationListener =
-			item.on('play-animation', (animationId: string) => {
-				console.log('PhaserItem play-animation', `${key}/${animationId}`);  // TODO remove
-				sprite.play(`${key}/${animationId}`);
-			});
-
-		scene.events.on('update', this.update, this);
+class PhaserItem extends PhaserAnimatedEntity {
+	// can probably just be a sprite
+	protected gameObject: Phaser.GameObjects.Container;
+	protected entity: Item;
+
+	constructor (
+		scene: Phaser.Scene,
+		entity: Item
+	) {
+		super(scene, entity, `item/${entity._stats.itemTypeId}`);
+
+		const translate = entity._translate;
+		this.gameObject = scene.add.container(
+			translate.x,
+			translate.y,
+			[ this.sprite ]
+		);
 	}
 
-	update (/*time: number, delta: number*/): void {
-
-		const item = this.item;
-		const container = item._pixiContainer;
-		const texture = item._pixiTexture;
-
-		if (item._destroyed || container._destroyed) {
-
-			item.off('hide', this.hide);
-			this.hide = null;
-
-			item.off('show', this.show);
-			this.show = null;
-
-			item.off('play-animation', this.playAnimationListener);
-			this.playAnimationListener = null;
-
-			this.scene.events.off('update', this.update, this);
-
-			this.sprite = null;
-
-			this.destroy();
-
-			return;
-		}
-
-		this.x = container.x;
-		this.y = container.y;
+	protected transform (data: {
```

Comment:
this and scale can probably move up to parent

Reply:
yup, I noticed it being used by projectile and unit too